nucypher/blockchain/eth/signers/hardware.py

```diff
@@ -0,0 +1,241 @@
+        """Defer account unlocking to the trezor, do not indicate application level unlocking logic."""
+        return True
+
+    @validate_checksum_address
+    def lock_account(self, account: str) -> bool:
+        """Defer account locking to the trezor, do not indicate application level unlocking logic."""
+        return True
+
+    @property
+    def accounts(self) -> List[str]:
+        """Returns a list of cached trezor checksum addresses from initial derivation."""
+        return list(self.__addresses.keys())
+
+    @handle_trezor_call
+    def sign_message(self, message: bytes, checksum_address: str) -> HexBytes:
+        """
+        Signs a message via the TREZOR ethereum sign_message API and returns
+        a named tuple containing the signature and the address used to sign it.
+        This method requires interaction between the TREZOR and the user.
+        """
+        # TODO: #2262 Implement Trezor Message Signing
+        hd_path = self.__get_address_path(checksum_address=checksum_address)
+        signed_message = trezorlib.ethereum.sign_message(self.__client, hd_path, message)
+        return HexBytes(signed_message.signature)
+
+    def sign_transaction(self,
+                         transaction_dict: dict,
+                         rlp_encoded: bool = True
+                         ) -> Union[HexBytes, Transaction]:
+        """
+        Sign a transaction with a trezor hardware wallet.
+
+        This method handles transaction validation, field formatting, signing,
+        and outgoing serialization.  Accepts a standard transaction dictionary as input,
+        and produces an RLP encoded raw signed transaction by default.
+
+        Internally the standard transaction dictionary is reformatted for trezor API consumption
+        via calls `trezorlib.client.ethereum.sign_tx`.
+
+        WARNING: This function returns a raw signed transaction which can be
+        broadcast by anyone with a connection to the ethereum network.
+
+        ***Treat pre-signed raw transactions produced by this function like money.***
+
+        """
+
+        # Consume the sender inside the transaction request's 'from field.
+        checksum_address = transaction_dict.pop('from')
+
+        # Format contract data field for both trezor and eth_account's Transaction
+        if transaction_dict.get('data') is not None:  # empty string is valid
+            transaction_dict['data'] = Web3.toBytes(HexBytes(transaction_dict['data']))
+
+        # Format transaction fields for Trezor, Lookup HD path, and Sign Transaction
+        # Leave the chain ID in tact for the trezor signing request:
+        # If `chain_id` is included, an EIP-155 transaction signature will be applied
+        # https://github.com/trezor/trezor-core/pull/311
+        if 'chainId' not in transaction_dict:
+            raise self.SignerError('Invalid EIP-155 transaction - "chain_id" field is missing in trezor signing request.')
```

Comment:
```suggestion
            raise self.SignerError('Invalid EIP-155 transaction - "chainId" field is missing in trezor signing request.')
```
```suggestion
            raise self.SignerError('Invalid EIP-155 transaction - "chain_id" field is missing in trezor signing request.')
```

Reply:
I think this is the only RFC pending before merging. Waiting for @KPrasch to confirm if the suggestion is correct, but I think it is.

Reply:
Trezor actually expects `chain_id` - despite this not being standard with `chainId`

Reply:
Oh right, we convert "chainId" -> "chain_id" for the trezor transaction format. I know we are trying to be proactive here by checking early, perhaps we should check this after the `trezor_transaction = self._format_transaction(transaction_dict=transaction_dict)` to make this clearer...?